yb-voyager/cmd/live_migration.go

```diff
@@ -16,42 +16,67 @@ limitations under the License.
 package cmd
 
 import (
-	"encoding/json"
 	"fmt"
-	"os"
-	"path/filepath"
+	"time"
 
 	log "github.com/sirupsen/logrus"
 
 	"github.com/yugabyte/yb-voyager/yb-voyager/src/tgtdb"
-	"github.com/yugabyte/yb-voyager/yb-voyager/src/utils"
 )
 
 func streamChanges() error {
-	queueFilePath := filepath.Join(exportDir, "data", "queue.json")
-	log.Infof("Streaming changes from %s", queueFilePath)
-	file, err := os.OpenFile(queueFilePath, os.O_CREATE, 0640)
-	if err != nil {
-		return fmt.Errorf("error opening file %s: %v", queueFilePath, err)
+	eventQueue := tgtdb.NewEventQueue(exportDir)
+	log.Infof("Streaming changes from %s", eventQueue.QueueDirPath)
+	for { // continuously get next segments to stream
+		segments, err := eventQueue.GetNextSegments()
```

Comment:
Code could look like: 
```
for {
    segment = eventQueue.getNextSegment()
    streamChangesForSegment(segment) // read all events in that segment
    // update state that segment has been complete. 
}
```

Reply:
> Why not deterministically just start with index=0, and when the file is done, roll over to the next segment?

In case of restarts, the initial segments will be already executed. So the first segments won't be `0` in restarts